src/oauth/helpers.ts

```diff
@@ -19,18 +19,36 @@ export type AccessGrant = {
   expiry: Date;
 };
 
+export function randomBytes(length: number): string {
+  return base64.fromArrayBuffer(
+    crypto.getRandomValues(new Uint8Array(length)).buffer as ArrayBuffer,
+    true,
+  );
+}
+
+export function generatePKCECodeVerifier() {
+  return randomBytes(32);
+}
```

Comment:
This is only used by our tests, normally the client will use a random value for the PKCE code verifier.